drivers/ethernet/eth_stellaris.c

```diff
@@ -175,38 +183,34 @@ static int eth_stellaris_rx_pkt(struct device *dev, struct net_pkt *pkt)
 
 		count = bytes_left - 4;
 		data = (u8_t *)&reg_val;
-		if (net_pkt_append(pkt, count, data, K_NO_WAIT) != count) {
-			return -1;
+		if (net_pkt_write_new(pkt, data, count)) {
+			goto error;
 		}
 
 		bytes_left -= 4;
 	}
 
-	return frame_len;
+	return pkt;
+
+error:
+	net_pkt_unref(pkt);
+
+	return NULL;
 }
 
 static void eth_stellaris_rx(struct device *dev)
 {
-	struct net_pkt *pkt = NULL;
 	struct eth_stellaris_runtime *dev_data = DEV_DATA(dev);
 	struct net_if *iface = dev_data->iface;
-	int frame_len, ret;
+	struct net_pkt *pkt;
 
-	/* Obtain the packet to be populated. */
-	pkt = net_pkt_get_reserve_rx(K_NO_WAIT);
+	pkt = eth_stellaris_rx_pkt(dev, iface);
 	if (!pkt) {
-		LOG_ERR("Could not allocate pkt");
+		LOG_ERR("Failed to read data");
```

Comment:
Minor thing, but I think we should say "Failed to write data" here.

Reply:
at this point we are actually trying to read data from the hw, sure we write it back to pkt. But it's same same I would say.